### PR TITLE
Codebreaker console app

### PR DESCRIPTION
**Pull request by DaniilYusha**

### CODE REVIEW

File: entities/console.rb [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+  def initialize
+    @output = OutputService.new
+    @registrator = RegistratorService.new
+    @decorator = Decorator.new
+    @statistics = Codebreaker::StatisticsService.new FILE_PATH
+    @error_exist = false
+    @output.introduction
+  end
+
```
Comment (artem247) [inline entities/console.rb:16]: I like it very much

File: modules/questioner.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+module Questioner
+  YES = 'y'
+  NO = 'n'
+  START_COMMAND = 'start'
+  RULES_COMMAND = 'rules'
+  STATS_COMMAND = 'stats'
+  EXIT_COMMAND = 'exit'
+  HINT_COMMAND = 'hint'
+
```
Comment (artem247) [inline modules/questioner.rb]: Nit: You can add i18n here if you're already using it elsewhere

File: entities/console.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  FILE_PATH = './db/statistics.yml'
+
+  def initialize
+    @output = OutputService.new
+    @registrator = RegistratorService.new
+    @decorator = Decorator.new
+    @statistics = Codebreaker::StatisticsService.new FILE_PATH
+    @error_exist = false
+    @output.introduction
```
Comment (Dimasfullxq) [inline entities/console.rb]: Imho in initialize should be only init data. Pls, move that from constructor.

File: entities/console.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      puts '=' * 25
+    end
+  end
+
+  def rules
+    system 'clear'
+    @output.show_rules
+  end
+
+  def hint
```
Comment (Dimasfullxq) [inline entities/console.rb]: I think this method should be in something kinda GameAdapter class.
Console can work with that class. Its gonna be lit 🔥.

**Review by Dimasfullxq — CHANGES_REQUESTED** [review]
I think would be cool if you added GameAdapter or GameService class.

**Review by artem247 — APPROVED** [review]
I think it's very good

File: entities/console.rb (hunk 1 above)
Comment (DaniilYusha) [reply entities/console.rb:16]: Thanks) but previous reviewer doesn't agree with us :smile:

File: modules/questioner.rb (hunk 2 above)
Comment (DaniilYusha) [reply modules/questioner.rb]: Good point, I think about it too

Comment (DaniilYusha) [reply modules/questioner.rb]: @artem247 Done

**Review by Dimasfullxq — APPROVED** [review]

**Review by katiakub — APPROVED** [review]
Everything is awesome!

File: entities/console.rb [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  def stats
+    system 'clear'
+    return output.no_stats unless File.exist? FILE_PATH
+
+    output.show_stats
+    decorated_stats = decorator.stats_beautify statistics.sort_statistics
+    decorated_stats.each_with_index do |user, index|
+      puts output.rating + index.next.to_s
+      user.each { |key, value| puts "#{key} #{value}" }
+      puts '=' * 25
```
Comment (Svatok) [inline entities/console.rb:38]: What is `25`? This is magic number?

**Review by Svatok — APPROVED** [review]
Please remove magic number

**Pull request merged**